api/pkg/service/catalog/catalog.go

```diff
@@ -78,3 +80,25 @@ func (s *service) Refresh(ctx context.Context, p *catalog.RefreshPayload) (*cata
 
 	return ret, nil
 }
+
+// List all Catalog
+func (s *service) List(ctx context.Context) (*catalog.ListResult, error) {
+
+	log := s.base.Logger(ctx)
+	db := s.base.DB(ctx)
+	var all []model.Catalog
+	if err := db.Find(&all).Error; err != nil {
+		log.Error(err)
+		return nil, internalError
+	}
+
+	res := &catalog.ListResult{
+		Data: []*catalog.Catalog{},
+	}
+
+	for _, c := range all {
+		res.Data = append(res.Data, &catalog.Catalog{ID: c.ID, Name: c.Name, Type: c.Type})
```

Comment:
may be we should add url also?

Reply:
if someone is integrating APIs then knowing which catalogs are in hub this API would be useful
I think we should expose it for users.. wdyt

Reply:
Yes, makes sense